### ISSUE TITLE
If an agent runs for too long a timeout exception is thrown

### ISSUE THREAD

**Issue by mlejva** [e2b-dev/e2b#16]
**Describe the bug**
I was generating an endpoint for Stripe checkout which requires relatively a lot of work. The agent was working correctly but after a while it timed out
```

app.listen(port, () =>DELETE /sessions/sc5lwio6-2dd8205d 204 - - 153.751 ms
[2023-04-07 21:56:47,524] ERROR in app: Exception on request POST /generate
Traceback (most recent call last):
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/quart/app.py", line 1650, in handle_request
    return await self.full_dispatch_request(request_context)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/quart/app.py", line 1675, in full_dispatch_request
    result = await self.handle_user_exception(error)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/quart/app.py", line 1107, in handle_user_exception
    raise error
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/quart/app.py", line 1673, in full_dispatch_request
    result = await self.dispatch_request(request_context)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/quart/app.py", line 1718, in dispatch_request
    return await self.ensure_async(handler)(**request_.view_args)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/app.py", line 84, in generate
    await cg.generate(
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/codegen/codegen.py", line 220, in generate
    await self._agent_executor.arun(
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chains/base.py", line 237, in arun
    return (await self.acall(kwargs))[self.output_keys[0]]
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chains/base.py", line 154, in acall
    raise e
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chains/base.py", line 148, in acall
    outputs = await self._acall(inputs)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/agents/agent.py", line 667, in _acall
    next_step_output = await self._atake_next_step(
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/codegen/agent/base.py", line 90, in _atake_next_step
    output = await self.agent.aplan(intermediate_steps, **inputs)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/agents/agent.py", line 305, in aplan
    action = await self._aget_next_action(full_inputs)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/agents/agent.py", line 260, in _aget_next_action
    full_output = await self.llm_chain.apredict(**full_inputs)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chains/llm.py", line 167, in apredict
    return (await self.acall(kwargs))[self.output_key]
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chains/base.py", line 154, in acall
    raise e
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chains/base.py", line 148, in acall
    outputs = await self._acall(inputs)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chains/llm.py", line 135, in _acall
    return (await self.aapply([inputs]))[0]
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chains/llm.py", line 123, in aapply
    response = await self.agenerate(input_list)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chains/llm.py", line 67, in agenerate
    return await self.llm.agenerate_prompt(prompts, stop)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chat_models/base.py", line 106, in agenerate_prompt
    raise e
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chat_models/base.py", line 100, in agenerate_prompt
    output = await self.agenerate(prompt_messages, stop=stop)
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chat_models/base.py", line 63, in agenerate
    results = await asyncio.gather(
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/langchain/chat_models/openai.py", line 287, in _agenerate
    async for stream_resp in await acompletion_with_retry(
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 230, in <genexpr>
    return (
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/openai/api_requestor.py", line 319, in wrap_resp
    async for r in resp:
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/openai/api_requestor.py", line 633, in <genexpr>
    return (
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/openai/api_requestor.py", line 114, in parse_stream_async
    async for line in rbody:
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/aiohttp/streams.py", line 35, in __anext__
    rv = await self.read_func()
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/aiohttp/streams.py", line 311, in readline
    return await self.readuntil()
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/aiohttp/streams.py", line 343, in readuntil
    await self._wait("readuntil")
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/aiohttp/streams.py", line 303, in _wait
    with self._timer:
  File "/Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/aiohttp/helpers.py", line 721, in __exit__
    raise asyncio.TimeoutError from None
asyncio.exceptions.TimeoutError
Executing <Task pending name='Task-18' coro=<ASGIHTTPConnection.handle_request() running at /Users/vasekmlejnsky/Developer/e2b/api-service/.venv/lib/python3.10/site-packages/quart/asgi.py:102> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py:429> cb=[_wait.<locals>._on_completion() at /usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/tasks.py:475] created at /usr/local/Cellar/python@3.10/3.10.8/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/tasks.py:636> took 0.394 seconds
[2023-04-07 21:56:47 +0200] [34713] [INFO] 127.0.0.1:61539 POST /generate 1.1 500 422915 214393211
```

**Expected behavior**
It shouldn't time out

**Screenshots**
<img width="1014" alt="Screen Shot 2023-04-07 at 9 59 30 PM" src="https://user-images.githubusercontent.com/5136688/230670826-64117e1d-05f7-4a08-9fe1-8a3f1038b90b.png">

![Screen Shot 2023-04-07 at 9 59 48 PM](https://user-images.githubusercontent.com/5136688/230671143-a73ffd65-9873-42c6-bbf4-59b7ee96667e.png)
![Screen Shot 2023-04-07 at 9 59 52 PM](https://user-images.githubusercontent.com/5136688/230671174-43f6df7f-ef89-4651-b621-97fb05a476b5.png)

**Desktop (please complete the following information):**
 - Chrome

**Comment by aodrasa** [comment]
Having the exact same issue (GPT-3.5). Tried running the endpoint multiple times and even with different endpoints - all end up timing out.

**Comment by ValentaTomas** [comment]
I think this is related to the OpenAI request timeout issue https://github.com/hwchase17/langchain/issues/1929#issue-1637686943.

**Comment by mlejva** [comment]
If the problem is on the OpenAI's end, this might be relevant for this issue https://platform.openai.com/docs/guides/rate-limits/overview

**Comment by ValentaTomas** [comment]
This should be fixed after merging #20 by increasing OpenAI timeout.
